crates/solvers/src/domain/liquidity/mod.rs

```diff
@@ -47,3 +53,80 @@ impl TokenPair {
         (self.0, self.1)
     }
 }
+
```

Comment:
This just moved verbatim from `weighted_product` as it is shared across liquidity types (notably Balancer-style weighted product pools and Curve-style stable pools).